snuba/replacer.py

```diff
@@ -414,7 +415,13 @@ def flush_batch(self, batch: Sequence[Replacement]) -> None:
 
             self.__replacer_processor.post_replacement(replacement, count)
 
-            self._check_timing_and_write_to_redis(replacement, start_time)
+            self._check_timing_and_write_to_redis(replacement, start_time.timestamp())
+
+            get_project_id = getattr(replacement, "get_project_id", None)
```

Comment:
Please do not go around the type checker.
The `Replacement` object does not have a `get_project_id` because it is higher level than an `ErrorReplacement`. 
So if the project id is needed outside the replacer processor, you will have to change the abstraciton and move the `get_project_id` method to the parent class. 

The fact the method is in the subclass but you have access to the parent class implies you are only supposed to use the methods from the parent class otherwise. Having the method in the subclass means you could receive a different type of replacement here that does not have `get_project_id` or that `get_project_id` could be removed from the replacement subclass. This code is not supposed to depend on it. 
Going around that abstraction fundamentally breaks the design.